src/Forms/TextField.php

```diff
@@ -16,6 +16,26 @@ class TextField extends FormField
 
     protected $schemaDataType = FormField::SCHEMA_DATA_TYPE_TEXT;
 
+    /**
+     * @var bool Whether the Tip UI should be rendered
+     */
+    protected $tipEnabled = false;
```

Comment:
I think it would make more sense to encapsulate this logic in a Tip interface/class and pass that around.

This would allow us to use regular setters/getters to interact with the FormField tip. Also, if we decide we need to display tips in a different context latter on, we'll have a nice unified API to do so.